lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -127,24 +127,28 @@ defmodule ExUnit.Runner do
     pending_tests =
       case process_max_failures(config, test_module) do
         :no ->
-          for test <- excluded_tests do
-            EM.test_started(config.manager, test)
-            EM.test_finished(config.manager, test)
-          end
-
           invalid_tests ++ excluded_tests
 
         {:reached, n} ->
-          Enum.take(invalid_tests, n)
+          Enum.take(invalid_tests, n) ++ excluded_tests
```

Comment:
The issue with this approach is that, if we have a formatter that also prints excluded_tests, it means that we will print N excluded tests after max_failures. If the goal is to account `excluded_tests` more reliably, then I would suggest to run all excluded_tests before everything else, before we call `spawn_module`. What do you think?

Reply:
We can print the excluded and skipped tests as soon as they are determined. I like that approach. Later in another loop we print the invalid ones.
But I fail to see how now "N excluded tests"  are printed, as I think only "N invalid tests" are printed

Reply:
Please have a look and tell me if the code now solves the issue that you mentioned regarding the N excluded tests